src/common/utils/navbar.ts

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from "react";
```

Comment:
move this to src/common/hooks/useXXX.ts, because its a custom hooks and have a state/reactive.